workspaces/optic-engine/src/interactions/visitors/diff.rs

```diff
@@ -197,26 +197,36 @@ impl RequestBodyVisitor<InteractionDiffResult> for DiffRequestBodyVisitor {
   fn visit(&mut self, interaction: &HttpInteraction, context: &RequestBodyVisitorContext) {
     if let Some(operation) = context.operation {
       let actual_content_type = &interaction.request.body.content_type;
+      let actual_body: Option<BodyDescriptor> = (&interaction.request.body.value).into();
       let (request_id, request_body_descriptor) = operation;
       //dbg!( actual_content_type);
       //dbg!(&request_body_descriptor);
-      match (&request_body_descriptor.body, actual_content_type) {
-        (None, None) => {
+      match (
+        &request_body_descriptor.body,
+        actual_content_type,
+        actual_body,
+      ) {
+        (None, None, _) => {
           self
             .visited_with_matched_content_types
             .insert(request_id.clone());
         }
-        (None, Some(content_type)) => {
+        (None, Some(content_type), None) => {
+          self
+            .visited_with_matched_content_types
+            .insert(request_id.clone());
+        }
+        (None, Some(content_type), Some(interaction_body)) => {
           self
             .visited_with_unmatched_content_types
             .insert(request_id.clone());
         }
```

Comment:
The difference between these two cases is definitely subtle, but written this way also makes sense. The interaction might have been sent with a content type, but in the end we're matching bodies to bodies. The content type is only really interesting if you want to compare 2 present bodies, a `None` matching `None` doesn't require that.

Nice one!